token-client/README.md

```diff
@@ -1,239 +1,306 @@
+`HttpClientFactory` creates an HTTP client that will make the requests to the corresponding Identity service.
+
+#### Autoconfigured tokenFlowHttpClient
+When using `spring-security` client library, a readily configured `CloseableHttpClient` is accessible via `tokenFlowHttpClient` Bean that uses the `HttpClientFactory` internally to set up the HTTP Client for token flows.
 
+#### Default HttpClientFactory
+The Token Client library includes a default implementation [DefaultHttpClientFactory](./src/main/java/com/sap/cloud/security/client/DefaultHttpClientFactory.java), of the [HttpClientFactory](./src/main/java/com/sap/cloud/security/client/HttpClientFactory.java) interface. 
+It creates a preconfigured [Apache HttpClient 4](https://hc.apache.org/httpcomponents-client-4.5.x/index.html) with the given [ClientIdentity](../java-api/src/main/java/com/sap/cloud/security/config/ClientIdentity.java) for the Identity service instance.
+
+To acquire the HTTP client, use the following code:
 ```java
-OAuth2ServiceConfigurationBuilder builder = OAuth2ServiceConfigurationBuilder.forService(Service.XSUAA);
-OAuth2ServiceConfiguration config = builder.withClientId(...)
-                                           .withClientSecret(...)
-                                           .withCertificate(...)
-                                           .withPrivateKey(...)
-                                           .withUrl(...)
-                                           .withCertUrl(...).build();
+CloseableHttpClient client = HttpClientFactory.createClient(ClientIdentity clientIdentity); // you can obtain ClientIdentity by invoking the OAuthServiceConfiguration.getClientIdentity() method
 ```
 
-#### Adapt to externally managed certificate
-If you want to use an **externally managed certificate** within your Java application, `OAuth2ServiceConfiguration` instance needs to be adapted with the external key:
+`DefaultHttpClientFactory` HTTP Clients are configured in the following way:
+- connection pool
+    - maximum of 200 total connections
+    - 50 connections per route.
+- connection and connection request timeouts -  5 seconds
+- socket timeout 30 seconds
+
+:information_source: These values are intended as an initial configuration, and you should monitor your application's performance and provide your own `HttpClientFactory` implementation, if you observe performance degradation.
+For more information, refer to the [Troubleshooting](#insufficient-performance-for-token-validations-or-token-flows) section.
+
+### Cache Configuration
+
+By default, the `OAuth2TokenService` implementations (`DefaultOAuth2TokenService` and `XsuaaOAuth2TokenService`) are caching tokens internally.
+By default up to 1000 tokens are cached for 10 minutes and the statistics are disabled.
+The Cache can be individually configured by configuring `TokenCacheConfiguration` class and providing it as a constructor parameter to `DefaultOAuth2TokenService` or `XsuaaOAuth2TokenService`.
```

Comment:
Would the cache actually be used though after this code? To be on the safe side, we could repeat at this place what you said in the beginning of the document: that you need to overwrite the default XsuaaTokenFlows bean if you want this cache object to actually be used. And perhaps add that for non Spring Boot cases, one needs to pass this token service to the XsuaaTokenFlows object that you use. If you see the constructor it should be clear but if the only thing you have seen about it, is this document, then it could save you some time fiddling around.